Controllers/PostController.cs

```diff
@@ -189,4 +189,28 @@ public IActionResult DeletePost(int id)
 
         return NoContent();
     }
+
+    [HttpGet("user/{id}")]
+    [Authorize]
+    public IActionResult GetPostsByUserId(int id)
+    {
+        List <Post> postsByUser = _dbContext
+            .Posts.Include(p => p.Author)
+            .Include(p => p.Comments)
+            .ThenInclude(c => c.Commenteer)
+            .ThenInclude(u => u.IdentityUser)
+            .Include(p => p.PostTags)
+            .Where(p => p.Id == id).ToList();
+
+        if (postsByUser == null)
+        {
+            return NotFound("no posts by this user");
+        }
+        if (postsByUser.Any(p => p.IsApproved == false))
+        {
+            return BadRequest("posts haven't been approved");
```

Comment:
This will return bad request if any post is unapproved, even if some of them are approved. Maybe include a check for isApproved in the `Where` to filter out unapproved posts instead.